libs/metagen/src/mdk_python/static/types.py.jinja

```diff
@@ -0,0 +1,25 @@
+import json
+from typing import List, Dict, Optional
+from dataclasses import dataclass, asdict, fields
+
+class Struct:
```

Comment:
Does the class provide any advantage or remove limitations?

Reply:
I tried `.pyi` first but it does not allow a good representation of union/either types for example nor enable hinting python dict when editing (turns out [box ](https://pypi.org/project/python-box/) is not part of the std python).
We can instead put structs and other complex types into classes, the downside is the manual serialization/deserialization work but since this is not designed to be manually edited, we can use codegen as much as we want for the tricky parts.

Reply:
Yes, `.ipy` is quite limiting. Dataclasses should be better than boxed struct indeed 👍 but imo != classes, that might be my confusion. Is there anything that requires to have this Struct as a class and instead uses data classes all the time?

Reply:
Not really, it does not have any fields so a simple class will do the job but I guess we can do it that way for consistency.